ios/ReactNativeCameraKit/CKCamera.m

```diff
@@ -13,6 +13,172 @@
 #import "CKCameraOverlayView.h"
 #import "CKMockPreview.h"
 
+NSString * NSStringFromInterfaceOrientation(UIInterfaceOrientation o){
+    if (o==UIInterfaceOrientationPortrait) {
+        return @"Portrait";
+    } else if (o==UIInterfaceOrientationLandscapeLeft){
+        return @"LandscapeLeft";
+    } else if (o==UIInterfaceOrientationLandscapeRight){
+        return @"LandscapeRight";
+    } else if (o==UIInterfaceOrientationPortraitUpsideDown){
+        return @"PortraitUpsideDown";
+    } else {
+        @throw @"unknown interface orientation";
+    }
+}
+
+AVCaptureVideoOrientation AVCaptureVideoOrientationFromInterfaceOrientation(UIInterfaceOrientation o){
+    if (o == UIInterfaceOrientationPortrait) {
+        return AVCaptureVideoOrientationPortrait;
+    } else if (o == UIInterfaceOrientationLandscapeLeft){
+        return AVCaptureVideoOrientationLandscapeLeft;
+    } else if (o == UIInterfaceOrientationLandscapeRight){
+        return AVCaptureVideoOrientationLandscapeRight;
+    } else if (o == UIInterfaceOrientationPortraitUpsideDown){
+        return AVCaptureVideoOrientationPortraitUpsideDown;
+    } else {
+        @throw @"unknown interface orientation";
+    }
+}
+
+@interface CKScannerOverlay: UIView
+@property (nonatomic,strong) UIView *topView, *leftSideView,*rightSideView,*bottomView;
+@property (nonatomic,strong) NSArray *cornerViews;
+@property (nonatomic,strong) UIView * dataReadingFrame;
+@property (nonatomic) CGFloat frameOffset;
+@property (nonatomic) CGFloat frameHeight;
+@property (nonatomic) UIView *scannerView;
+@property (nonatomic, strong) UIColor *laserColor;
+@property (nonatomic, strong) UIColor *frameColor;
+@property (nonatomic, assign) BOOL shouldAnimating;
+@end
+
+@implementation CKScannerOverlay
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    self = [super initWithFrame:frame];
+    if (self) {
+        self.frameOffset = 30;
+        self.frameHeight = 200;
+
+        self.dataReadingFrame = [UIView new];
```

Comment:
`new` is equivalent to alloc, init, but it may have some unintended consequences. For UIView, it's probably fine to use. But I would like to get in the habit of using `[[UIView alloc] init]`, it flows better with  `[[UIView alloc] initWithFrame: ]`, and avoids any unexpected consequences if the class has overridden` new`. It's also more conventional.